src/components/smart_objects/include/smart_objects/smart_object.h

```diff
@@ -680,7 +680,8 @@ class SmartObject FINAL {
    */
```

Comment:
Could add a description for the new `allow_unknown_enums` param to the function brief